gluten-core/src/main/scala/org/apache/gluten/extension/columnar/TransformSingleNode.scala

```diff
@@ -303,7 +303,15 @@ object TransformOthers {
   class ReplaceSingleNode() extends LogLevelUtil with Logging {
 
     def doReplace(p: SparkPlan): SparkPlan = {
-      val plan = p
+      val plan = p match {
+        case plan: FileSourceScanExec
+            if plan.relation.fileFormat.getClass.getSimpleName == "ArrowCSVFileFormat" =>
+          val arrowScan = ArrowFileSourceScanExec(plan)
+          TransformHints.tagNotTransformable(arrowScan, "Arrow scan cannot transform")
+          return arrowScan
+        case p => p
+      }
```

Comment:
The code is placed in `gluten-core` but relies on code in `backend-velox`. Would you like to check whether we can add a rule via backend API?

Probably through `BackendsApiManager.getSparkPlanExecApiInstance.genExtendedColumnarValidationRules` ?

Reply:
Can I change it in a following PR? This PR is too big